core-bundle/src/Resources/contao/modules/ModulePassword.php

```diff
@@ -183,9 +183,15 @@ protected function setNewPassword()
 	{
 		$optIn = System::getContainer()->get('contao.opt_in');
 
-		// Find an unconfirmed token with only one related record
-		if ((!$optInToken = $optIn->find(Input::get('token'))) || !$optInToken->isValid() || \count($arrRelated = $optInToken->getRelatedRecords()) != 1 || key($arrRelated) != 'tl_member' || \count($arrIds = current($arrRelated)) != 1 || (!$objMember = MemberModel::findByPk($arrIds[0])))
-		{
+		// Find an unconfirmed token with only one related tl_member record
+		if (
+			(!$optInToken = $optIn->find(Input::get('token')))
+			|| !$optInToken->isValid()
+			|| !isset($optInToken->getRelatedRecords()['tl_member'])
+			|| \count($arrRelated = $optInToken->getRelatedRecords()['tl_member']) != 1
```

Comment:
```suggestion
			|| \count($arrRelated = $optInToken->getRelatedRecords()) != 1
```

Otherwise this would be a behavior change. Previously `count($optInToken->getRelatedRecords())` had to be `1` but with your change there could be several related records as long as there is only one `tl_member` property.

Reply:
Yes, that's the point of this PR. It doesn't matter how many related records there are. It matters that there's only one `tl_member` :)

Reply:
No, I don‘t think so. We are looking for a token with one single related record that holds the member ID and nothing else here.